paperqa/types.py

```diff
@@ -110,8 +110,20 @@ class Text(Embeddable):
         description="Source document this text chunk originates from.",
     )
 
+    def __eq__(self, other) -> bool:
+        if not isinstance(other, type(self)):
+            return NotImplemented
+        # We ignore the embedding since the embedding can:
+        # - Be lazily acquired, or not used (depending on settings)
+        # - Get ditched when serializing a text for an HTTP request
+        return (
+            self.name == other.name
+            and self.text == other.text
+            and self.doc == other.doc
```

Comment:
Is this one implemented?

Reply:
Wait - why doesn't this infinitely recurse? 

if you hit the doc == doc line, then that calls the text == text in the docs, which will call the doc == doc line, etc.